src/server/client.ts

```diff
@@ -885,77 +887,104 @@ export class Auth0Client {
           ...existingSession.internal
         }
       });
-    } else {
-      const req = reqOrSession as PagesRouterRequest | NextRequest;
 
-      if (!sessionData) {
-        throw new Error("The session data is missing.");
-      }
+      return;
+    }
 
-      if (req instanceof NextRequest && res instanceof NextResponse) {
-        // middleware usage
-        const existingSession = await this.getSession(req);
+    const req = reqOrSession as PagesRouterRequest | NextRequest | Request;
 
-        if (!existingSession) {
-          throw new Error("The user is not authenticated.");
-        }
+    if (!sessionData) {
+      throw new Error("The session data is missing.");
+    }
 
-        await this.sessionStore.set(req.cookies, res.cookies, {
-          ...sessionData,
-          internal: {
-            ...existingSession.internal
-          }
-        });
-      } else {
-        // pages router usage
-        const existingSession = await this.getSession(
-          req as PagesRouterRequest
-        );
+    if (req instanceof NextRequest && res instanceof NextResponse) {
+      // middleware usage
+      const existingSession = await this.getSession(req);
+
+      if (!existingSession) {
+        throw new Error("The user is not authenticated.");
+      }
 
-        if (!existingSession) {
-          throw new Error("The user is not authenticated.");
+      await this.sessionStore.set(req.cookies, res.cookies, {
+        ...sessionData,
+        internal: {
+          ...existingSession.internal
         }
+      });
 
-        const resHeaders = new Headers();
-        const resCookies = new ResponseCookies(resHeaders);
-        const updatedSession = sessionData as SessionData;
-        const reqCookies = this.createRequestCookies(req as PagesRouterRequest);
-        const pagesRouterRes = res as PagesRouterResponse;
+      return;
+    }
 
-        await this.sessionStore.set(reqCookies, resCookies, {
-          ...updatedSession,
-          internal: {
-            ...existingSession.internal
-          }
-        });
+    /*
+     * Next 16 proxy.ts path
+     * In Next 16 the function in proxy.ts gets a plain Web Request.
+     * We need to turn it into a NextRequest so that sessionStore can read cookies, then write back to the NextResponse.
+     */
+    if (req instanceof Request && res instanceof NextResponse) {
```

Comment:
If we do `toNextRequest` at the beginning, we do not need this additional branch and can rely on the code above to handle `NextRequest`